File: solutions/julia/hamming/1/hamming.jl
"Calculates hamming distance of two DNA strands"
function distance(a, b)
    if length(a)!=length(b)
        throw(ArgumentError("Strands must be of the same length"))
    end
    delta = 0
    for i=1:length(a) 
        delta += a[i] != b[i] ? 1 : 0
    end
    delta
end

File: solutions/dart/hamming/1/lib/hamming.dart
class Hamming {
  int distance(String a, String b){
    if(a.length != b.length){
      throw ArgumentError('strands must be of equal length');
    }
    int dist = 0;
    for(var i=0; i<a.length; i++){
      dist += a[i]==b[i] ? 0 : 1;
    }
    return dist;
  }
}
